<launch>
  <!-- define all args -->
  <arg name="gui" default="false" />
  <arg name="vis" default="true" />

  <!-- set all parameters -->
  <param name="rate" value="30" />
  <param name="use_gui" value="$(arg gui)"/>
  <param name="robot_description" textfile="$(find pendulum_3d)/urdf/pendulum.urdf" />

  <node name="joy_node" pkg="joy" type="joy_node" >
    <!-- <remap from="/joy" to="/board_joy" /> -->
  </node>

  <!-- launch pendulum state nodes -->
  <node name="robot_state_publisher" pkg="robot_state_publisher" type="state_publisher"/>

  <!-- launch pendulum simulator -->
  <node name="pendulum_simulator" pkg="pendulum_3d" type="pendulum_3d_simulator_corr.py" output="screen" respawn="false" />

  <!-- launch rviz -->
  <group if="$(arg vis)">
    <node pkg="rviz" type="rviz" respawn="false" name="rviz"
      args="-d $(find pendulum_3d)/launch/pendulum_3d.rviz" />
  </group>

</launch>
